### PR TITLE
Set working directory to current directory in Start-Job

### PR DESCRIPTION
**Pull request by iSazonov**
# PR Summary

As approved by PowerShell-Committee in #10673 set a working directory to current directory in Start-Job cmdlet.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: https://github.com/MicrosoftDocs/PowerShell-Docs/issues/5012
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by iSazonov** [comment]
@rjmholt @TylerLeonhardt I hope this doesn't break any tool.

File: src/System.Management.Automation/engine/remoting/commands/StartJob.cs [hunk 1]
```diff
@@ -608,6 +608,17 @@ protected override void BeginProcessing()
                     ThrowTerminatingError(errorRecord);
```
Comment (PoshChan) [inline src/System.Management.Automation/engine/remoting/commands/StartJob.cs]: @iSazonov, your last commit had 1 failures in `PowerShell-CI-static-analysis`
Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/learning-powershell/powershell-beginners-guide.md.https://examples.oreilly.com/9780596528492/ should work
```powershell
retry of URL failed with error: Connection timed out
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 117
117:                                     throw "retry of URL failed with error: $($_.Exception.Message)"

```

**Comment by TylerLeonhardt** [comment]
> I hope this doesn't break any tool

The PowerShell extension doesn't use `Start-Job` for anything and this wouldn't effect EditorSyntax since the syntax isn't changing.

I'm not 100% sure but doubt that PSScriptAnalyzer would be affected because I don't see why it would use `Start-Job` when it's a binary module.

**Review by SteveL-MSFT — APPROVED** [review]
LGTM

**Comment by SteveL-MSFT** [comment]
@PoshChan please retry static

**Comment by PoshChan** [comment]
@SteveL-MSFT, successfully started retry of `PowerShell-CI-static-analysis`

**Review by PaulHigin — APPROVED** [review]
LGTM

File: src/System.Management.Automation/engine/remoting/commands/StartJob.cs (hunk 1 above)
Comment (PoshChan) [inline src/System.Management.Automation/engine/remoting/commands/StartJob.cs]: @iSazonov, your last commit had 2 failures in `PowerShell-CI-static-analysis`
Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/learning-powershell/powershell-beginners-guide.md.https://examples.oreilly.com/9780596528492/ should work
```powershell
retry of URL failed with error: Connection timed out
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 117
117:                                     throw "retry of URL failed with error: $($_.Exception.Message)"

```
Verify Markdown Links.Verify links in /home/vsts/work/1/s/docs/learning-powershell/powershell-beginners-guide.md.https://examples.oreilly.com/9780596528492/ should work
```powershell
retry of URL failed with error: Connection timed out
at <ScriptBlock>, /home/vsts/work/1/s/test/common/markdown/markdown-link.tests.ps1: line 117
117:                                     throw "retry of URL failed with error: $($_.Exception.Message)"

```

**Pull request merged**